apps/gk_field.c

```diff
@@ -330,12 +337,90 @@ gk_field_new(struct gkyl_gk *gk, struct gkyl_gyrokinetic_app *app)
     }
     // Deflated Poisson solve is performed on range assuming decomposition is *only* in z.
     f->flr_op = gkyl_deflated_fem_poisson_new(app->grid, app->basis_on_dev, app->basis,
-      app->local, app->local, f->flr_rhoSq_sum, f->flr_kSq, flr_bc, app->use_gpu);
+      app->local, app->local, f->flr_rhoSq_sum, f->flr_kSq, flr_bc, NULL, app->use_gpu);
   }
 
+    // twist-and-shift boundary condition for phi and skin surface from ghost to impose phi periodicity at z=-pi
+  if (f->gkfield_id == GKYL_GK_FIELD_ES_IWL){
+    gk_field_add_TSBC_and_SSFG_updaters(app,f);
+  }
+  
   return f;
 }
 
+void
+gk_field_add_TSBC_and_SSFG_updaters(struct gkyl_gyrokinetic_app *app, struct gk_field *f)
```

Comment:
this function doesn't need to be public, i.e. it should be removed from the header file and declared as static